system/include/emscripten/threading.h

```diff
@@ -36,19 +36,20 @@ void emscripten_force_num_logical_cores(int cores);
 
 // If the given memory address contains value val, puts the calling thread to
 // sleep waiting for that address to be notified.
-int emscripten_futex_wait(volatile void/*uint32_t*/ *addr, uint32_t val, double maxWaitMilliseconds);
+int emscripten_futex_wait(volatile void/*uint32_t*/ *addr __attribute__((nonnull)), uint32_t val, double maxWaitMilliseconds);
 
 // Wakes the given number of threads waiting on a location. Pass count ==
 // INT_MAX to wake all waiters on that location.
-int emscripten_futex_wake(volatile void/*uint32_t*/ *addr, int count);
+// Returns -EINVAL if addr is null.
+int emscripten_futex_wake(volatile void/*uint32_t*/ * _Nonnull addr, int count);
```

Comment:
Why the difference between wake and wait?

Reply:
See the earlier comment, wrote there about the closer differences between these two.

`emscripten_futex_wake` tests and returns `-EINVAL` if its `addr` is null. So it could not be made `__attribute__((nonnull))`. but must use weaker `_Nonnull`.